### PR TITLE
sql: extract frontend code into separate package

### PR DESCRIPTION
**Pull request by gabor**
short version:
we moved `/public/app/features/plugins/sql` to `/packages/grafana-sql`, and adjusted the imports.

long version:
there is sql-related frontend-code that is used by multiple datasources: postgres, mysql, mssql and influxdb (in some modes).

the shared code currently lives in this folder: `/public/app/features/plugins/sql`.

we are working on decoupling these datasources from core-grafana, so this PR moves these files into a new package, at `/packages/grafana-sql`. 

this package will stay "virtual", we are not going to npm-release it for now.

the PR touches many files, but the changes are:
- the moving of the files from `/public/app/features/plugins/sql`. the files themselves did not change at all, they were only moved
- `package.json` and `tsconfig.json` for the new `/packages/grafana-sql` package
- changed imports everywhere

### CODE REVIEW

**Comment by bohandley** [comment]
Nice! We are doing a similar thing with Prometheus [here](https://github.com/grafana/grafana/pull/77762). It is nice that all the sql code was already decomposed and in one place.

**Review by aangelisc — APPROVED** [review]
LGTM, effective change for the decoupling effort

**Comment by gabor** [comment]
i could not test the influxdb-related part, could someone from metrics-squad try that out pls? ( pinging @itsmylife 😄 )

File: .betterer.results [hunk 1]
```diff
@@ -594,6 +594,28 @@ exports[`better eslint`] = {
...
+    "packages/grafana-sql/src/components/query-editor-raw/RawEditor.tsx:5381": [
+      [0, 0, 0, "Styles should be written using objects.", "0"],
+      [0, 0, 0, "Styles should be written using objects.", "1"]
+    ],
+    "packages/grafana-sql/src/components/visual-query-builder/AwesomeQueryBuilder.tsx:5381": [
+      [0, 0, 0, "Unexpected any. Specify a different type.", "0"]
+    ],
+    "packages/grafana-sql/src/components/visual-query-builder/SQLWhereRow.tsx:5381": [
+      [0, 0, 0, "Do not use any type assertions.", "0"]
+    ],
```
Comment (jackw) [inline .betterer.results:618]: nit: would be nice to clean these up. probably best left for a follow up PR.

**Review by jackw — APPROVED** [review]
I've not been through all the code changes but from a frontend-ops/yarn workspace perspective this looks good to me. 👍 

Probably worth waiting for someone that knows the code better to comment on any specifics there.

File: .betterer.results (hunk 1 above)
Comment (gabor) [reply .betterer.results:618]: i agree, i just wanted to keep the moved files unchanged, to make the pr easier to review.

**Review by itsmylife — APPROVED** [review]
influxdb sql changes are working. 🚀

File: packages/grafana-sql/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "@grafana/experimental": "1.7.6",
+    "@grafana/runtime": "workspace:*",
+    "@grafana/ui": "workspace:*",
+    "@react-awesome-query-builder/ui": "6.4.2",
+    "@types/lodash": "4.14.202",
+    "@types/react-virtualized-auto-sizer": "1.0.4",
+    "@types/uuid": "9.0.7",
+    "immutable": "4.3.4",
+    "lodash": "4.17.21",
+    "react-use": "17.4.0",
```
[the remark this answers is not in the document]
Comment (gabor) [reply packages/grafana-sql/package.json:28]: good catch! i think this should not be a problem, but i'll sync this to the main-package's version in a sepsrate PR 👍 
(this one got into a conflict anyway 😿 )

**Pull request merged**